src/platform/nrfconnect/wifi/WiFiManager.h

```diff
@@ -205,6 +214,19 @@ class WiFiManager
     static void PostConnectivityStatusChange(ConnectivityChange changeType);
     static void SendRouterSolicitation(System::Layer * layer, void * param);
 
+    // Connection Recovery feature
+    // This feature allows re-scanning and re-connecting the connection to the known network after
+    // a reboot or when a connection is lost. The following attempts will occur with increasing interval.
```

Comment:
I am afraid that this will not be applied in case the connection is lost in general. We would need to trigger this mechanism in DisconnectHandler and then we would need to rely on the supplicant to always send the disconnection event reliably. So this implementation IMO applies to the Matter fail safe mechanism solely.